garden_ai/client.py

```diff
@@ -362,6 +364,126 @@ def register_pipeline(self, pipeline: Pipeline, container_uuid: str) -> str:
         local_data.put_local_pipeline(registered)
         return func_uuid
 
+    def create_minimal_step(
+        self,
+        *,
+        model_name: str,
+        local_path: str,
+        flavor: str,
+        input_type: type,
+        output_type: type,
+        requirements_file: str = "requirements.txt",
+    ) -> Step:
+        with open(requirements_file, "r") as f:
+            pip_reqs = [line.strip() for line in f.readlines()]
+
+        local_model = LocalModel(
+            model_name=model_name,
+            flavor=flavor,
+            extra_pip_requirements=pip_reqs,
+            local_path=local_path,
+            user_email=self.get_email(),
+        )
+
+        registered_model = self.register_model(local_model)
+
+        @step
+        def run_inference(
+            input_arg: input_type, model=Model(registered_model.full_name)
+        ) -> output_type:
+            return model.predict(input_arg)
+
+        return run_inference
+
+    def publish_minimal_pipeline(
+        self,
+        _step: Step,
+        garden_doi: str,
+        *,
+        authors: List[str],
+        tags: List[str],
```

Comment:
There will potentially be even more metadata fields like this in the near future. (The stuff Allison and Chase are up to.) Should we keep growing this argument list for every field?

Reply:
The logical response being "No," but stick with me for a second. The obvious response is to setup some **kwargs and let Python do the rest; but then you say "what if the user provides nonsense?" My solution, however, is not to make a schema, which will cost dev hours down the line as users fail to use it properly anyway and bugs inevitably arise, and rather to put our foot down and say this is a minimal example in every sense of the word--and if something arises that is so critical it becomes a minimum requirement, then the trouble to add an argument to this function is negligible.

Reply:
just to chime in on this, I think that you are basically right when you say the obvious thing to do is setup some **kwargs and let ~python~ pydantic do the rest. 

See for example the way you use the `create_pipeline` method here -- when `create_pipeline` was written, we had no concept of the `short_name` field. Cut to now, a world where you need `short_name` to satisfy the `RegisteredPipeline` schema. Notice how you were able to easily specify `short_name=model_name` at the call site but we didn't need to modify the definition of `create_pipeline` in tandem with the changing schema? 

tl;dr object/dataclass/schema/whatever is already the place where we can/ought to encode "what's the absolute minimum"